src/main/java/im/status/keycard/KeycardApplet.java

```diff
@@ -338,9 +368,26 @@ private void processInit(APDU apdu) {
     apdu.setIncomingAndReceive();
 
     if (selectingApplet()) {
+      short off = 0;
+
+      // Add the secure channel public key to the output info
       apduBuffer[0] = TLV_PUB_KEY;
       apduBuffer[1] = (byte) secureChannel.copyPublicKey(apduBuffer, (short) 2);
-      apdu.setOutgoingAndSend((short) 0, (short)(apduBuffer[1] + 2));
+      off += 2;
+      off += apduBuffer[1];
+
+      // Get the instance UID. This is a unique identifier for the installation and
+      // cannot be updated after the app is installed. It is used to make the certificates.
+      apduBuffer[off] = TLV_UID;
+      off += 1;
```

Comment:
Does this imply that the TLV_UID is one byte? That seems awfully small for a UID... 

Sanity check or better explanation required.

Reply:
As we just discussed, anything starting with `TLV_` is a single byte id for the data field.